src/lib/maplibre/controls/CustomControl/Icon.svelte

```diff
@@ -0,0 +1,20 @@
+<script lang="ts">
+	import type { Snippet } from 'svelte';
+	let {
+		children,
+		onclick,
+		oncontextmenu,
+		className
+	}: { children?: Snippet; className: string; onclick?: () => void; oncontextmenu?: () => void } = $props();
```

Comment:
I'd recommend importing `MouseEventHandler` from `'svelte/elements'` and using it as shown below:

```suggestion
		onclick?: MouseEventHandler<HTMLButtonElement>;
		oncontextmenu?: MouseEventHandler<HTMLButtonElement>;
	} = $props();
```